app/controllers/sample_types_controller.rb

```diff
@@ -12,7 +12,9 @@ class SampleTypesController < ApplicationController
   before_action :find_assets, only: [:index]
   before_action :auth_to_create, only: %i[new create]
   before_action :project_membership_required, only: %i[create new select filter_for_select]
+  before_action :old_attributes, only: %i[update]
 
+  after_action :update_sample_json_metadata, only: :update
```

Comment:
What would happen if 2 updates were made to a sample type in a short time?

I think you may need a mechanism for "locking" the sample type to further changes until the update job has completed.

Reply:
Didn't know that was possible. Has it already been done somewhere else in the source code?